ada_feeding/ada_feeding/ada_watchdog_listener.py

```diff
@@ -0,0 +1,97 @@
+"""
+This module defines the ADAWatchdogListener class, which listens to the watchdog
+topic and can be queried to determine if the watchdog is still alive.
+"""
+
+# Third-party imports
+from diagnostic_msgs.msg import DiagnosticArray, DiagnosticStatus
+from rcl_interfaces.msg import ParameterDescriptor, ParameterType
+from rclpy.duration import Duration
+from rclpy.node import Node
+from rclpy.time import Time
+
+
+# pylint: disable=too-few-public-methods
+# This class only needs one public method to check if the watchdog is ok.
+class ADAWatchdogListener:
+    """
+    The ADAWatchdogListener class listens to the watchdog topic and can be queried
+    to determine if the watchdog is still alive.
+    """
+
+    def __init__(self, node: Node):
```

Comment:
Not necessary, but consider adding an event-based API to this as well. i.e. init with an optional callback function (that can be changed later). And once the watchdog fails, call that function with the Diagnostic.

In a separate function to set/change the callback function, don't allow it to happen if the watchdog has already failed.